src/map/lua/luautils.cpp

```diff
@@ -1685,10 +1685,11 @@ namespace luautils
     {
         TracyZoneScoped;
 
-        auto name = PChar->m_moghouseID ? "Residential_Area" : (const char*)zoneutils::GetZone(PChar->loc.destination)->GetName();
+        auto name     = PChar->m_moghouseID ? "Residential_Area" : (const char*)zoneutils::GetZone(PChar->loc.destination)->GetName();
+        auto filename = fmt::format("./scripts/zones/{}/Zone.lua", name);
 
         auto onZoneInFramework = lua["xi"]["globals"]["interaction"]["interaction_global"]["onZoneIn"];
-        auto onZoneIn = lua["xi"]["zones"][name]["Zone"]["onZoneIn"];
+        auto onZoneIn = GetCacheEntryFromFilename(filename)["onZoneIn"];
```

Comment:
Assuming this is to support modules loading beforehand?  Just wondering, since we seem to load these pretty early